packages/modular-views.macro/src/index.macro.js

```
@@ -3,45 +3,51 @@ const path = require('path');
-    .readdirSync(viewsDirectoryPath, { withFileTypes: true })
-    // Get individual view directories.
-    .filter((entry) => entry.isDirectory())
-    // Get view `package.json`s.
-    .map((dir) =>
-      fs.readJSONSync(path.join(viewsDirectoryPath, dir.name, 'package.json')),
-    )
-    .filter(
-      (packageJson) =>
-        // only chose the ones explicitly marked as views
-        packageJson.modular?.type === 'view' &&
-        // Remove views which are marked as private (and therefore are not published yet.)
-        packageJson.private !== true,
-    )
-    // Get package names.
-    .map((packageJson) => packageJson.name);
+const viewsDirectoryPath = path.join(getModularRoot(), 'packages');
 
-  return `(() => ({
+const packageNames = fs
+  .readdirSync(viewsDirectoryPath, { withFileTypes: true })
+  // Get individual view directories.
+  .filter((entry) => entry.isDirectory())
+  // Get view `package.json`s.
+  .map((dir) =>
+    fs.readJSONSync(path.join(viewsDirectoryPath, dir.name, 'package.json')),
+  )
+  .filter(
+    (packageJson) =>
+      // only chose the ones explicitly marked as views
+      packageJson.modular?.type === 'view' &&
+      // Remove views which are marked as private (and therefore are not published yet.)
+      packageJson.private !== true,
+  )
+  // Get package names.
+  .map((packageJson) => packageJson.name);
+
+const viewsMap = `({
   ${packageNames
     .map(
-      (packageName) => `'${packageName}': lazy(() => import('${packageName}'))`,
+      (packageName) =>
+        `'${packageName}': __lazy__(() => import('${packageName}'))`,
     )
     .join(',\n  ')}
-}))`;
-}
+})`;
 
 function views({ references, babel }) {
-  // eslint-disable-next-line
-  const { types: t } = babel;
-  // TODO: add a react import
-  references.default.forEach((ref) => {
-    ref.replaceWithSourceString(
-      generateViewMap(path.join(getModularRoot(), 'packages')),
+  if (references.default.length > 1) {
+    console.warn(
```

this should probably throw an error